src/main/kotlin/com/zepben/auth/client/AuthProviderConfig.kt

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright 2024 Zeppelin Bend Pty Ltd
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+package com.zepben.auth.client
+
+import com.zepben.auth.common.AuthException
+import com.zepben.auth.common.AuthMethod
+import com.zepben.auth.common.StatusCode
+import io.vertx.core.json.DecodeException
+import io.vertx.core.json.Json
+import io.vertx.core.json.JsonObject
+import java.io.IOException
+import java.net.URI
+import java.net.http.HttpClient
+import java.net.http.HttpRequest
+import java.net.http.HttpResponse
+
+data class ProviderDetails(
+    val issClaim: String,
+    val tokenEndpoint: String,
+    val jwkUrl: String
+)
+
+data class AuthProviderConfig(
+    val issuer: String,
+    val audience: String,
+    val authType: AuthMethod = AuthMethod.OAUTH,
+    val protocol: String = "https",
+    val providerDetails: ProviderDetails = ProviderDetails("", "", "")
+)
+
+// Fetch info from the provider
+// Used in EWB/EAS, DON'T MAKE PRIVATE!
+fun fetchProviderDetails(
+    issuer: String,
+    protocol: String = "https",
+    client: HttpClient = HttpClient.newBuilder().sslContext(SSLContextUtils.allTrustingSSLContext()).build(),
+    handler: HttpResponse.BodyHandler<String> = HttpResponse.BodyHandlers.ofString()
+): ProviderDetails {
+    val issuerURL = "${issuer.takeIf { it.startsWith("http") } ?: "$protocol://$issuer"}/v2.0/.well-known/openid-configuration"
+    val response = client.send(HttpRequest.newBuilder().uri(URI(issuerURL)).GET().build(), handler)
+    if (response.statusCode() == StatusCode.OK.code) {
+        try {
+            val authConfigJson = Json.decodeValue(response.body()) as JsonObject
+            return ProviderDetails(
+                tokenEndpoint = authConfigJson.getString("token_endpoint", ""),
+                jwkUrl = authConfigJson.getString("jwks_uri", ""),
+                issClaim = authConfigJson.getString("issuer", "")
+            )
```

Comment:
it would be a problem for sure. I'm not overly worried though, as we're talking Auth0 and Entra, if we can't get oidc config I bet we have bigger issues.